disnake/appinfo.py

```diff
@@ -402,6 +448,154 @@ def user_install_type_config(self) -> Optional[InstallTypeConfiguration]:
+            The custom installation url for this application.
+        role_connections_verification_url: Optional[:class:`str`]
+            The application's role connection verification entry point,
+            which when configured will render the app as a verification method
+            in the guild role verification configuration.
+        interactions_endpoint_url: Optional[:class:`str`]
+            The application's interactions endpoint URL.
+        icon: Optional[|resource_type|]
+            Update the application's icon asset, if any.
+        cover_image: Optional[|resource_type|]
+            Update the cover_image for rich presence integrations.
+        guild_install_type_config: Optional[:class:`InstallTypeConfiguration`]
+            The guild installation type configuration for this application.
+        user_install_type_config: Optional[:class:`InstallTypeConfiguration`]
+            The user installation type configuration for this application.
+        event_webhooks_url: Optional[:class:`str`]
+            The application's event webhooks URL.
+        event_webhooks_status: :class:`ApplicationEventWebhookStatus`
+            The application's event webhooks status.
+        event_webhook_types: Optional[List]
+            The application's event webhook types. See `webhook event types <https://discord.com/developers/docs/events/webhook-events#event-types>`_
+            for a list of valid events.
+
+        Raises
+        ------
+        HTTPException
+            Editing the application information failed.
+
+        Returns
+        -------
+        :class:`.AppInfo`
+            The new application information.
+        """
+        fields: EditAppInfoPayload = {}
+
+        if install_params is not MISSING:
+            fields["install_params"] = (
+                install_params.to_dict() if install_params is not None else None
+            )
+
+        if guild_install_type_config is not MISSING or user_install_type_config is not MISSING:
+            integration_types_config: Dict[
+                ApplicationIntegrationTypeLiteral, ApplicationIntegrationTypeConfigurationPayload
+            ] = {}
+
+            if guild_install_type_config is not MISSING:
+                integration_types_config[0] = (
+                    guild_install_type_config.to_dict() if guild_install_type_config else {}
+                )
+            elif self.guild_install_type_config:
+                integration_types_config[0] = self.guild_install_type_config.to_dict()
+
+            if user_install_type_config is not MISSING:
+                integration_types_config[1] = (
+                    user_install_type_config.to_dict() if user_install_type_config else {}
+                )
+            elif self.user_install_type_config:
+                integration_types_config[1] = self.user_install_type_config.to_dict()
+
+            fields["integration_types_config"] = integration_types_config
```

Comment:
```suggestion
        if guild_install_type_config is not MISSING or user_install_type_config is not MISSING:
            integration_types_config: Dict[str, ApplicationIntegrationTypeConfigurationPayload] = {}

            if guild_install_type_config is MISSING:
                guild_install_type_config = self.guild_install_type_config
            if guild_install_type_config:
                integration_types_config["0"] = guild_install_type_config.to_dict()

            if user_install_type_config is MISSING:
                user_install_type_config = self.user_install_type_config
            if user_install_type_config:
                integration_types_config["1"] = user_install_type_config.to_dict()

            fields["integration_types_config"] = integration_types_config
```
simplifies it a tiny bit. Additionally:
- ensures object keys are strings (`json` seemingly stringifies them automatically, `orjson` raises)
- turns a `None` parameter value into a missing key instead of a `{}` value for the key (since there'd otherwise be no way of disabling an install type)